flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -260,4 +268,57 @@ public Optional<OperationFlowsView> getLastFlowOperationByResourceCrn(String res
         List<FlowLog> flowLogs = flowLogDBService.getLatestFlowLogsByCrnInFlowChain(resourceCrn);
         return flowOperationStatisticsService.createOperationResponse(resourceCrn, flowLogs);
     }
+
+    public List<FlowLogResponse> getFlowLogsByIds(List<String> flowIds, Pageable pageable) {
+        LOGGER.info("Getting flow logs by flow ids list {}", flowIds);
+        return flowLogDBService.getFlowLogsByFlowIdsCreatedDesc(new HashSet<>(flowIds), pageable)
+                .stream().map(flowLog -> flowLogConverter.convert(flowLog)).collect(toList());
+    }
+
+    public List<FlowCheckResponse> getFlowChainsByChainIds(List<String> chainIds, Pageable pageable) {
+        LOGGER.info("Getting flow check responses by flow check ids list {}", chainIds);
+        Map<String, FlowChainLog> flowChainsMap = flowChainLogService
+                .findAllByFlowChainIdInOrderByCreatedDesc(new HashSet<>(chainIds), pageable)
+                .stream().collect(Collectors.toMap(FlowChainLog::getFlowChainId, Function.identity()));
+
+        // GETTING RELATED FLOW CHAIN LOGS
+        Map<String, List<FlowChainLog>> relatedFlowChainsMap = new HashMap<>();
+        Map<String, String> parentMap = new HashMap<>();
+        flowChainsMap.forEach((key, flowChainLog) -> {
+            List<FlowChainLog> relatedFlowChainLogs = flowChainLogService.getRelatedFlowChainLogs(List.of(flowChainLog));
+            relatedFlowChainsMap.put(key, relatedFlowChainLogs);
+            relatedFlowChainLogs.forEach(log -> parentMap.put(log.getFlowChainId(), key));
+            relatedFlowChainLogs.add(flowChainLog);
```

Comment:
this part is pretty confusing, what happens and why. maybe it could be clarified with some variable rename,
eg: `key` -> `parentFlowChainId`
`parentMap` -> `parentChainIdByChainIdMap` or whatever which reflects that the key is a chaind id and the  value is the parent chain id